store/src/java/com/zimbra/cs/service/mail/RestoreContacts.java

```diff
@@ -0,0 +1,147 @@
+                    }
+                    String url = realm + getLocalHost() + "/service/home/"
+                        + mbox.getAccount().getName() + "/?" + UserServlet.QP_FMT + "=tgz&"
+                        + UserServlet.QP_TYPES + "=contact&" + MimeConstants.P_CHARSET + "=UTF-8";
+                    CookieStore cookieStore = new BasicCookieStore();
+                    AuthToken authToken = octxt.getAuthToken();
+                    BasicClientCookie cookie = null;
+                    try {
+                        cookie = new BasicClientCookie(ZimbraCookie.COOKIE_ZM_AUTH_TOKEN, authToken.getEncoded());
+                        cookie.setPath("/");
+                        cookie.setDomain(mbox.getAccount().getDomainName());
+                        cookieStore.addCookie(cookie);
+                    } catch (AuthTokenException e) {
+                        throw ServiceException.FAILURE("Failed to get authentication token", e);
+                    }
+                    File file = new File(FileBlobStore.getBlobPath(mbox, doc.getId(),
+                        doc.getSavedSequence(), Short.valueOf(doc.getLocator())));
+                    if (!file.exists()) {
+                        throw ServiceException
+                            .INVALID_REQUEST("File does not exist: " + contactBackupFileName, null);
+                    }
+                    ZimbraLog.contact.debug("Backup file found. Restoring contacts in %s.",
+                        contactBackupFileName);
+                    httpResponse = httpPostBackup(file, url, cookieStore);
+                    break;
+                }
+            }
+        }
+        if (!mailItemFound) {
+            throw ServiceException.INVALID_REQUEST("No such file: " + contactBackupFileName, null);
+        }
+        ContactRestoreStatus status = httpResponse.getStatusLine().getStatusCode() == HttpStatus.OK_200 ? ContactRestoreStatus.SUCCESS : ContactRestoreStatus.FAILURE;
+        if ("SUCCESS".equals(status)) {
+            ZimbraLog.contact.debug("Restore operation for %s completed successfully",
+                contactBackupFileName);
+        } else {
+            ZimbraLog.contact.info("Restore operation for %s failed. Http respose status: %s",
+                contactBackupFileName, httpResponse.getStatusLine());
+        }
+        response.setStatus(status.name());
+        return zsc.jaxbToElement(response);
+    }
+
+    public static HttpResponse httpPostBackup(File file, String url, CookieStore cookieStore) throws ServiceException {
+        HttpResponse httpResponse = null;
+        HttpClient http = HttpClientBuilder.create().setDefaultCookieStore(cookieStore)
+            .build();
+        HttpPost post = new HttpPost(url);
+        MultipartEntity multipart = new MultipartEntity();
+        multipart.addPart("file", new FileBody(file));
+        post.setEntity(multipart);
+        try {
+            httpResponse = http.execute(post);
```

Comment:
Apologies for commenting on closed PR. Just had a though on this one.
Will this upload file? If yes, that means we are uploading a file, which is already there on server.